tests/utils.go

```diff
@@ -0,0 +1,270 @@
+	param[0]["gas"] = "0x200000"
+
+	rpcRes := Call(t, "eth_sendTransaction", param)
+
+	var hash hexutil.Bytes
+	err := json.Unmarshal(rpcRes.Result, &hash)
+	require.NoError(t, err)
+
+	receipt := WaitForReceipt(t, hash)
+	require.NotNil(t, receipt, "transaction failed")
+	require.Equal(t, "0x1", receipt["status"].(string))
+
+	return hash, receipt
+}
+
+func DeployTestContractWithFunction(t *testing.T, addr []byte) hexutil.Bytes {
+	// pragma solidity ^0.5.1;
+
+	// contract Test {
+	//     event Hello(uint256 indexed world);
+	//     event TestEvent(uint256 indexed a, uint256 indexed b);
+
+	//     uint256 myStorage;
+
+	//     constructor() public {
+	//         emit Hello(17);
+	//     }
+
+	//     function test(uint256 a, uint256 b) public {
+	//         myStorage = a;
+	//         emit TestEvent(a, b);
+	//     }
+	// }
+
+	bytecode := "0x608060405234801561001057600080fd5b5060117f775a94827b8fd9b519d36cd827093c664f93347070a554f65e4a6f56cd73889860405160405180910390a260d08061004d6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c8063eb8ac92114602d575b600080fd5b606060048036036040811015604157600080fd5b8101908080359060200190929190803590602001909291905050506062565b005b8160008190555080827ff3ca124a697ba07e8c5e80bebcfcc48991fc16a63170e8a9206e30508960d00360405160405180910390a3505056fea265627a7a723158201d94d2187aaf3a6790527b615fcc40970febf0385fa6d72a2344848ebd0df3e964736f6c63430005110032"
+
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", addr)
+	param[0]["data"] = bytecode
+	param[0]["gas"] = "0x200000"
+
+	rpcRes := Call(t, "eth_sendTransaction", param)
+
+	var hash hexutil.Bytes
+	err := json.Unmarshal(rpcRes.Result, &hash)
+	require.NoError(t, err)
+
+	receipt := WaitForReceipt(t, hash)
+	require.NotNil(t, receipt, "transaction failed")
+	require.Equal(t, "0x1", receipt["status"].(string))
+
+	return hash
+}
+
+//nolint
+func GetTransactionReceipt(t *testing.T, hash hexutil.Bytes) map[string]interface{} {
+	param := []string{hash.String()}
+	rpcRes := Call(t, "eth_getTransactionReceipt", param)
+
+	receipt := make(map[string]interface{})
+	err := json.Unmarshal(rpcRes.Result, &receipt)
+	require.NoError(t, err)
+
+	return receipt
+}
+
+func WaitForReceipt(t *testing.T, hash hexutil.Bytes) map[string]interface{} {
+	for i := 0; i < 12; i++ {
+		receipt := GetTransactionReceipt(t, hash)
+		if receipt != nil {
+			return receipt
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	return nil
+}
+
+func GetNonce(t *testing.T, block string) hexutil.Uint64 {
+	from, err := GetAddress()
+	if err != nil {
+		fmt.Printf("failed to get account: %s\n", err)
+		os.Exit(1)
+	}
+
+	param := []interface{}{hexutil.Bytes(from), block}
+	rpcRes := Call(t, "eth_getTransactionCount", param)
+
+	var nonce hexutil.Uint64
+	err = json.Unmarshal(rpcRes.Result, &nonce)
+	require.NoError(t, err)
+	return nonce
+}
+
+func UnlockAllAccounts(t *testing.T) bool {
```

Comment:
no need to return a bool here
```suggestion
func UnlockAllAccounts(t *testing.T) {
```

Reply:
then i will return `error`. that sounds okay?

Reply:
nvm i see your suggested changes!